### ISSUE TITLE
JMX remote connection not working 

### ISSUE THREAD

**Issue by hongshaoyang** [quarkusio/quarkus#19009]
### Describe the bug

1. On GraalVM EE 21.1.0 + Quarkus, I tried to connect to a JMX connection by adding 
com.sun.management.jmxremote to java opts
2. Connect to the JMX port specified (e.g. `-Dcom.sun.management.jmxremote.port=9010`) in VisualVM
3. VisualVM can't connect via that JMX port.

I've also tried using JMC (JDK Mission Control) and JConsole (https://docs.oracle.com/en/java/javase/11/management/using-jconsole.html). Both can't connect via remote JMX connection.

Tried `./mvnw compile quarkus:dev` and `java -jar target/quarkus-app/quarkus-run.jar` and both can't work.

### Expected behavior

I would like to monitor CPU and memory of Quarkus app compared to vanilla Java app. For other app, I can connect to remote JMX connection.  

I should be able to connect via remote JMX connection to Quarkus app by specifying 
```
-Dcom.sun.management.jmxremote
-Dcom.sun.management.jmxremote.port=9010
```
... etc other com.sun.management.jmxremote options.

### Output of `uname -a` or `ver`

Darwin hongsy.local 19.6.0 Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64

### Output of `java -version`

```
java version "11.0.11" 2021-04-20 LTS
Java(TM) SE Runtime Environment GraalVM EE 21.1.0 (build 11.0.11+9-LTS-jvmci-21.1-b05)
Java HotSpot(TM) 64-Bit Server VM GraalVM EE 21.1.0 (build 11.0.11+9-LTS-jvmci-21.1-b05, mixed mode, sharing)
```

### Quarkus version or git rev

Quarkus 1.13.6.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /Users/xxx/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 11.0.11, vendor: Oracle Corporation, runtime: /Library/Java/JavaVirtualMachines/graalvm-ee-java11-21.1.0/Contents/Home
Default locale: en_SG, platform encoding: US-ASCII
OS name: "mac os x", version: "10.15.7", arch: "x86_64", family: "mac"
```

### Additional information

Java opts:
```
-Dcom.sun.management.jmxremote
-Dcom.sun.management.jmxremote.port=9010
-Dcom.sun.management.jmxremote.rmi.port=9010
-Dcom.sun.management.jmxremote.local.only=false
-Dcom.sun.management.jmxremote.authenticate=false
-Dcom.sun.management.jmxremote.ssl=false
```

Application Properties
```
%dev.quarkus.log.console.json=false
%test.quarkus.log.console.json=false
quarkus.native.additional-build-args=--language:js
quarkus.log.category."io.quarkus.arc.processor".level=DEBUG
quarkus.package.output-directory=app
quarkus.http.port=9000
```

**Comment by hongshaoyang** [comment]
Possibly related to #6943 where

**Comment by hongshaoyang** [comment]
Gentle bump on this issue, could anyone help to triage this issue?

Possibly related to #6943 where it is also related to JMX

**Comment by Sanne** [comment]
We don't support JMX; we actually intentionally disable it in many place.

**Comment by hongshaoyang** [comment]
Thanks @Sanne

would be good to update in the Quarkus docs that it doesn't support JMX.

**Comment by Sanne** [comment]
Fair enough; do you have a suggestion about where this should be written? Where would you have expected to find it?

**Comment by hongshaoyang** [comment]
@Sanne The [Measuring Performance](https://quarkus.io/guides/performance-measure) or [Micrometer Metrics](https://quarkus.io/guides/micrometer) guides look like a good place to get started, maybe a small info note will help. 

My team and I were trying out a Quarkus POC to compare its performance with our current Java stack, and we attempted to use JMX to get performance metrics, but failed.

**Comment by mhagnumdw** [comment]
I was a little confused. Does this also apply for Quarkus 2 + OpenJDK 17?

I read this post which shows how to enable the JMX remote: https://quarkus.io/blog/monitoring-quarkus-jvm-mode-with-cryostat/

I'm starting Keycloak (Keycloak 21.1.1 on JVM (powered by Quarkus 2.13.7.Final)) activating jmxremote, but the port doesn't stay up.

**Comment by rdas22** [comment]
Hi,
Even I am facing the port issue as @mhagnumdw mentioned for the same setup. (Keycloak 21.1.2.4.6 + Quarkus 2.13.8.Final)

Can anyone confirm if JMX is supported for that version?

Additionally, while researching on this, I found a few references which mentioned that some other process could be binding to that port before jmx can utilize it or multiple instances could be running for the same app. However, I checked and verified that its not the case.

Here's what I did:

Before starting the docker container, I checked the port which I assigned for jmx and nothing was running on it. After starting it, got BindException: Address already in use for that port.

Here are the JMX args that I am passing:
`      JAVA_TOOL_OPTIONS: >
        -Dcom.sun.management.jmxremote
        -Dcom.sun.management.jmxremote.port=9090
        -Dcom.sun.management.jmxremote.rmi.port=9090
        -Djava.rmi.server.hostname=0.0.0.0
        -Dcom.sun.management.jmxremote.ssl=false
        -Dcom.sun.management.jmxremote.authenticate=false
        -Dcom.sun.management.jmxremote.local.only=false`

and exposing the ports like:

`    ports:
      - "9090:9090"`

Exception:
`
Error: JMX connector server communication error: service:jmx:rmi://d94229cf370f:9090
jdk.internal.agent.AgentConfigurationError: java.rmi.server.ExportException: Port already in use: 9090; nested exception is: 
java.net.BindException: Address already in use
at jdk.management.agent/sun.management.jmxremote.ConnectorBootstrap.exportMBeanServer(ConnectorBootstrap.java:843)
at jdk.management.agent/sun.management.jmxremote.ConnectorBootstrap.startRemoteConnectorServer(ConnectorBootstrap.java:480)
at jdk.management.agent/jdk.internal.agent.Agent.startAgent(Agent.java:447)
at jdk.management.agent/jdk.internal.agent.Agent.startAgent(Agent.java:599)
Caused by: java.rmi.server.ExportException: Port already in use: 9090; nested exception is: 
java.net.BindException: Address already in use
at java.rmi/sun.rmi.transport.tcp.TCPTransport.listen(TCPTransport.java:346)
at java.rmi/sun.rmi.transport.tcp.TCPTransport.exportObject(TCPTransport.java:243)
at java.rmi/sun.rmi.transport.tcp.TCPEndpoint.exportObject(TCPEndpoint.java:415)
at java.rmi/sun.rmi.transport.LiveRef.exportObject(LiveRef.java:147)
at java.rmi/sun.rmi.server.UnicastServerRef.exportObject(UnicastServerRef.java:235)
at jdk.management.agent/sun.management.jmxremote.ConnectorBootstrap$PermanentExporter.exportObject(ConnectorBootstrap.java:205)
at java.management.rmi/javax.management.remote.rmi.RMIJRMPServerImpl.export(RMIJRMPServerImpl.java:153)
at java.management.rmi/javax.management.remote.rmi.RMIJRMPServerImpl.export(RMIJRMPServerImpl.java:138)
at java.management.rmi/javax.management.remote.rmi.RMIConnectorServer.start(RMIConnectorServer.java:453)
at jdk.management.agent/sun.management.jmxremote.ConnectorBootstrap.exportMBeanServer(ConnectorBootstrap.java:839)
`
